src/main/java/teammates/common/util/Version.java

```diff
@@ -0,0 +1,193 @@
+     * Those without number will be null.
+     * 
+     * If the version has more than 3 numbers, the first number will be major, the second number 
+     * will be minor and the rest will be patch.
+     * 
+     * For example: 
+     * version = 15
+     * major = 15, minor = null and patch = null
+     * 
+     * version = 15.01
+     * major = 15, minor = 1 and patch = null
+     * 
+     * version = 15.01.03
+     * major = 15, minor = 1 and patch = "03"
+     * 
+     * version = 15.01.03.01
+     * major = 15, minor = 1 and patch = "03.01"
+     * 
+     */
+    
+    
+    /**
+     * Creates a new instance of Version from string.
+     * It accepts either XX-XX-XXXXX or XX.XX.XXXX format.
+     */
+    public Version(String versionInString) {
+        String[] list;
+        if (versionInString.contains("-")) {
+            list = versionInString.split("-", 3);   // split into at most 3 parts
+        } else {
+            list = versionInString.split(".", 3);
+        }
+        if (list.length > 0) {
+            major = Integer.parseInt(list[0]);
+        }
+        if (list.length > 1) {
+            minor = Integer.parseInt(list[1]);
+        }
+        if (list.length > 2) {
+            patch = list[2];
+        }
+    }
+    
+    /**
+     * Compares by string representation.
+     */
+    public boolean equals(Object anotherVersion) {
+        return toString().equals(anotherVersion.toString());
+    }
+    
+    /**
+     * Gets hash code for this version.
+     */
+    public int hashCode() {
+        return toString().hashCode();
+    }
+    
+    /**
+     * Converts Version to String in format XX.XX.XXXX
+     */
+    public String toString() {
+        String version = "";
+        if (major != null) {
+            version += String.valueOf(major);
+        }
+        if (minor != null) {
+            String minorInString = String.valueOf(minor);
+            if (minorInString.length() < 2) {
+                minorInString = "0" + minorInString;
+            }
+            version += "." + minorInString;
+        }
+        if (patch != null) {
+            version += "." + patch;
+        }
+        return version;
+    }
+    
+    /**
+     * Converts to String in format XX-XX-XXXX
+     */
+    public String toStringForQuery() {
+        return toString().replace('.', '-');
+    }
+    
+    /**
+     * Gets all available versions.
+     */
+    public static List<Version> getAvailableVersions() {
+        List<String> versionListInString = new ArrayList<String>(modulesService.getVersions(null)); // null == default module
+        List<Version> versionList = new ArrayList<Version>();
+        for(String versionInString : versionListInString) {
+            versionList.add(new Version(versionInString));
+        }
+        Collections.sort(versionList, new VersionComparator());
+        return versionList;
+    }
```

Comment:
This method looks like it belongs in a `LogReader` class or a `GaeApi` class. It's fine to do the processing of the version info in this class but reading it from GAE doesn't belong in this class.